File: ios/Snag/Models/SnagControl.swift
import Foundation

public struct SnagControl: Codable, Sendable {
    public var type: String
    
    // Handshake
    public var deviceId: String?
    public var authMode: String?
    public var encryptedPayload: Data?
    public var encryptedNonce: Data?
    
    public var appInfo: SnagAppInfo?
    public var shouldStreamLogs: Bool?
    
    public init(type: String, 
                appInfo: SnagAppInfo? = nil, 
                shouldStreamLogs: Bool? = nil, 
                deviceId: String? = nil,
                authMode: String? = nil,
                encryptedPayload: Data? = nil,
                encryptedNonce: Data? = nil) {
        self.type = type
        self.appInfo = appInfo
        self.shouldStreamLogs = shouldStreamLogs
        self.deviceId = deviceId
        self.authMode = authMode
        self.encryptedPayload = encryptedPayload
        self.encryptedNonce = encryptedNonce
    }
}

File: android/snag/src/main/java/com/snag/models/SnagControl.kt
package com.snag.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SnagControl(
    @SerialName("type")
    val type: String, // "appInfoRequest", "appInfoResponse", "logStreamingControl", "logStreamingStatusRequest", "logStreamingStatusResponse"
    @SerialName("appInfo")
    val appInfo: SnagAppInfo? = null,
    @SerialName("shouldStreamLogs")
    val shouldStreamLogs: Boolean? = null,
    
    // Handshake
    @SerialName("deviceId")
    val deviceId: String? = null,
    @SerialName("authMode")
    val authMode: String? = null,
    
    // Encryption (Base64 encoded)
    @SerialName("encryptedPayload")
    val encryptedPayload: String? = null, // Base64
    @SerialName("encryptedNonce")
    val encryptedNonce: String? = null // Base64
)
